permuta/MeshPatt.py

```diff
@@ -338,45 +331,83 @@ def add_point(self, box, shade_dir=DIR_NONE, safe=True):
 
-    def add_increase(self,box):
+    def add_increase(self, pos):
         """Adds an increasing pattern (0, 1) into the given coordinate.
 
         Args:
-            box: tuple
-                The coordinate of the box to insert (0, 1) into.
+            pos: tuple
+                The coordinate of the position to insert (0, 1) into.
 
         Returns: <permuta.MeshPatt>
-            The new pattern with the increased pattern inserted into box.
+            The new pattern with the increased pattern inserted into pos.
 
         Examples:
             >>> MeshPatt((0,)).add_increase((0, 0))
             MeshPatt(Perm((0, 1, 2)), frozenset())
             >>> MeshPatt((0, 1, 2), [(1, 0), (2, 1), (3, 2)]).add_increase((2, 0))
             MeshPatt(Perm((2, 3, 0, 1, 4)), frozenset({(1, 2), (5, 4), (3, 3), (2, 3), (4, 3), (1, 0), (1, 1)}))
         """
-        x, y = box
+        x, y = pos
         return self.add_point((x, y)).add_point((x + 1,y + 1))
 
-    def add_decrease(self,box):
+    def add_decrease(self, pos):
         """Adds an decreasing pattern (1, 0) into the given coordinate.
 
         Args:
-            box: tuple
-                The coordinate of the box to insert (1, 0) into.
+            pos: tuple
+                The coordinate of the position to insert (1, 0) into.
 
         Returns: <permuta.MeshPatt>
-            The new pattern with the decreasing pattern inserted into box.
+            The new pattern with the decreasing pattern inserted into pos.
 
         Examples:
             >>> MeshPatt((0,)).add_decrease((0, 0))
             MeshPatt(Perm((1, 0, 2)), frozenset())
             >>> MeshPatt((0, 1, 2), [(1, 0), (2, 1), (3, 2)]).add_decrease((2, 0))
             MeshPatt(Perm((2, 3, 1, 0, 4)), frozenset({(1, 2), (5, 4), (3, 3), (2, 3), (4, 3), (1, 0), (1, 1)}))
         """
-        x, y = box
+        x, y = pos
         return self.add_point((x, y)).add_point((x + 1, y))
 
     #
+    # Occurrence/Avoidance/Containment methods
+    #
+
+    def occurrences_in(self, perm):
+        """Find all indices of occurrences of self in perm.
+
+        Args:
+            self:
+                The mesh pattern whose occurrences are to be found.
+            perm: <permuta.Perm>
+                The permutation to search for occurrences in.
+
+        Yields: numbers.Integral
+            The indices of the occurrences of self in perm. Each yielded element
+            l is a list of integer indices of the permutation perm such that
+            self.pattern == permuta.Perm.to_standard([perm[i] for i in l])
+            and that no element not in the occurrence falls within self.shading.
+        """
+        # TODO: Implement all nice
+        indices = list(range(len(perm)))
+        for candidate_indices in itertools.combinations(indices, len(self)):
+            candidate = [perm[index] for index in candidate_indices]
+            if Perm.to_standard(candidate) != self.pattern:
+                continue
```

Comment:
This should at least use the occurrence code of Perm. We don't want this piece of code to be this dumb at the master branch, and I say that having written this code myself.